### PR TITLE
Additional telemetry support for Panduit SmartPDU/Smart rack

### PR DESCRIPTION
**Pull request by uppalk1**
* Add Sensor data telemetry support

### CODE REVIEW

**Comment by tldavies** [comment]
Please fix the hound errors.

File: lib/utils/metrics/snmp-pdu-sensor-status.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    SnmpPduSensorMetric.prototype.collectSensorData = function() {
+        var self = this;
+        if (this.nodeType === 'sinetica') {
+            return Promise.all([
+                    self._collectSineticaSensorData(['HAWK-I2-MIB::ipTHATable']),
+                    self._collectSineticaSensorData(['HAWK-I2-MIB::ipContTable'])
+                ]
+                )
+                .then(function (results) {
+                    for (var i = (results.length)-1; i >= 0; i-- ) {
```
Comment (yyscamper) [inline lib/utils/metrics/snmp-pdu-sensor-status.js]: The function [_.remove](http://devdocs.io/lodash~4/index#remove) is better here

File: lib/utils/metrics/snmp-pdu-sensor-status.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        var self = this;
+        var snmpQueryType = 'bulkwalk';
+        return self.snmptool.collectHostSnmp(pduOids, {
+            snmpQueryType: snmpQueryType,
+               isSequential: true
+        })
+        .then(function (pduOidsResult) {
+            return pduOidsResult[0];
+
+        }).catch(function(err) {
```
Comment (yyscamper) [inline lib/utils/metrics/snmp-pdu-sensor-status.js]: This `catch` is just want to ignore any error while collecting sensor data?

File: lib/utils/metrics/snmp-pdu-sensor-status.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                return;
+            });
+    };
+
+    SnmpPduSensorMetric.prototype._calculateSineticaSensorData = function(sensorData, source) {
```
Comment (yyscamper) [inline lib/utils/metrics/snmp-pdu-sensor-status.js]: Please paste some example data here, so it help other user to understand the code.

File: lib/utils/metrics/snmp-pdu-sensor-status.js (hunk 2 above)
Comment (uppalk1) [inline lib/utils/metrics/snmp-pdu-sensor-status.js]: Yes, want to make sure the code executes the other sensor commands while it logs errors on the failed ones.

**Comment by keedya** [comment]
:+1:

File: lib/utils/metrics/snmp-pdu-sensor-status.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        }
+        return out;
+    };
+
+    SnmpPduSensorMetric.prototype._collectSineticaSensorData = function(pduOids) {
+        var self = this;
+        var snmpQueryType = 'bulkwalk';
+        return self.snmptool.collectHostSnmp(pduOids, {
+            snmpQueryType: snmpQueryType,
+               isSequential: true
```
Comment (yyscamper) [inline lib/utils/metrics/snmp-pdu-sensor-status.js]: indent error

File: lib/utils/metrics/snmp-pdu-sensor-status.js (hunk 2 above)
Comment (yyscamper) [inline lib/utils/metrics/snmp-pdu-sensor-status.js]: OK

**Comment by yyscamper** [comment]
:+1: after fix indent

File: lib/utils/metrics/snmp-pdu-sensor-status.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    };
+
+    SnmpPduSensorMetric.prototype.collectSensorData = function() {
+        var self = this;
+        if (this.nodeType === 'sinetica') {
+            return Promise.all([
+                    self._collectSineticaSensorData(['HAWK-I2-MIB::ipTHATable']),
+                    self._collectSineticaSensorData(['HAWK-I2-MIB::ipContTable'])
+                ]
+                )
```
Comment (brianparry) [inline lib/utils/metrics/snmp-pdu-sensor-status.js]: Indentation error?

**Comment by uppalk1** [comment]
@brianparry

**Pull request merged**